searchlight/tests/unit/test_glance_metadefs_plugin.py

```diff
@@ -0,0 +1,370 @@
+                               description='desc3'),
+        ]
+
+        self.namespaces[0].objects = [self.objects[0]]
+        self.namespaces[1].objects = self.objects[1:]
+
+    def _create_resource_types(self):
+        self.resource_types = [
+            _db_resource_type_fixture(name=RESOURCE_TYPE1,
+                                      protected=False),
+            _db_resource_type_fixture(name=RESOURCE_TYPE2,
+                                      protected=False),
+            _db_resource_type_fixture(name=RESOURCE_TYPE3,
+                                      protected=True),
+        ]
+
+    def _create_namespace_resource_types(self):
+        self.namespace_resource_types = [
+            _db_namespace_resource_type_fixture(
+                prefix='p1',
+                name=self.resource_types[0].name),
+            _db_namespace_resource_type_fixture(
+                prefix='p2',
+                name=self.resource_types[1].name),
+            _db_namespace_resource_type_fixture(
+                prefix='p2',
+                name=self.resource_types[2].name),
+        ]
+        self.namespaces[0].resource_types = self.namespace_resource_types[:1]
+        self.namespaces[1].resource_types = self.namespace_resource_types[1:]
+
+    def _create_tags(self):
+        self.tags = [
+            _db_resource_type_fixture(name=TAG1),
+            _db_resource_type_fixture(name=TAG2),
+            _db_resource_type_fixture(name=TAG3),
+        ]
+        self.namespaces[0].tags = self.tags[:1]
+        self.namespaces[1].tags = self.tags[1:]
+
+    @unittest.skip("Skipping metadefs")
+    def test_index_name(self):
+        self.assertEqual('glance', self.plugin.get_index_name())
+
+    @unittest.skip("Skipping metadefs")
+    def test_document_type(self):
+        self.assertEqual('metadef', self.plugin.get_document_type())
+
+    @unittest.skip("Skipping metadefs")
+    def test_namespace_serialize(self):
+        metadef_namespace = self.namespaces[0]
+        expected = {
+            'namespace': 'namespace1',
+            'display_name': '1',
+            'description': 'desc1',
+            'visibility': 'private',
+            'protected': True,
+            'owner': '6838eb7b-6ded-434a-882c-b344c77fe8df'
+        }
+        serialized = self.plugin.serialize_namespace(metadef_namespace)
+        self.assertEqual(expected, serialized)
+
+    @unittest.skip("Skipping metadefs")
+    def test_object_serialize(self):
+        metadef_object = self.objects[0]
+        expected = {
+            'name': 'Object1',
+            'description': 'desc1',
+            'properties': [{
+                'default': 'value1',
+                'enum': ['value1', 'value2'],
+                'property': 'property1',
+                'type': 'string'
+            }]
+        }
+        serialized = self.plugin.serialize_object(metadef_object)
+        self.assertEqual(expected, serialized)
+
+    @unittest.skip("Skipping metadefs")
```

Comment:
Yeah - the plugin isn't implemented yet (old implementation relied on direct database access). I skipped them rather than deleting them; intending to add support back in shortly.